src/lib/hooks/use-on-screen.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState, MutableRefObject } from 'react'
+
+/**
+ * Given a `ref`, returns a `boolean` indicating whether the `ref.current`
+ * element is visible on screen.
+ *
+ * From https://usehooks.com/useOnScreen/
+ */
+function useOnScreen(ref: MutableRefObject<HTMLElement>): boolean {
```

Comment:
💭 non-blocker: if I am understanding correctly, this is doing a lot of what you get from framer-motion `useInView` hook? https://www.framer.com/motion/use-in-view/ Checking to see if an element is within the viewport?

Reply:
ah true yeah, great catch! i'll try swapping in `useInView` 👍

Reply:
✅ 🙌 swapped it in, works great! Thanks for the suggestion 💖

Reply:
Not sure if its a 1to1 mapping of `useOnScreen` but was curious.